docs/docs/preprocessing-external-images.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Preprocessing External Images
+---
+
+Gatsby allows powerful image processing features using the [`Sharp`](https://github.com/lovell/sharp/) library to automatically process images to be performant, with features like lazy-loading. That said, this only works if the image is a `File` node in the GraphQL layer.
+
+If you want the same functionality for files that are remotely hosted online and not located in your Git repo, [`gatsby-source-filesystem`](/packages/gatsby-source-filesystem/) has an API called `createRemoteFileNode` to solve this.
+
+This guide will show you how to use the `createRemoteFileNode` process and get the same benefits of gatsby-transformer-sharp with externally sourced images.
+
+## Setup
+
+A use case that this technique can support is if you want to create a featured image in a blog post with an image sourced from a URL out on the web, instead of a local file. This could be hosted somewhere like Imgur, S3, or anywhere on the internet.
+
+Given a sample post:
+
+```markdown
+---
+title: My first blogpost!
+featuredImgUrl: https://images.unsplash.com/photo-1560237731-890b122a9b6c
+featuredImgAlt: Mountains with a starry sky
+---
+
+Hello World
+```
+
+You can use a custom Frontmatter field for the URL of the featured image you want to pull down and use as part of the site.
+
+By default, this is a string value as you haven't told Gatsby yet how to interpret it. However, you can add some code into `gatsby-node.js` to modify it.
+
+## Gatsby Node
+
+In your `gatsby-node.js` file, you can do some processing to create file nodes for the custom `featuredImgUrl` Frontmatter field.
```

Comment:
This should introduce what schema customization is and why it's necessary here before the code example, otherwise it comes straight out of left field and could overwhelm the reader.

Reply:
Done!